### PR TITLE
Enable Long tap gesture for files and folders on all screens

### PR DESCRIPTION
**Pull request by ankitgoyalgloballogic**
#REF: #MOBILEAPPS-1953

### CODE REVIEW

File: ContentApp/PresentationLayer/Components/ListComponent/ListComponentViewController.swift [hunk 1]
```diff
@@ -397,6 +404,14 @@ extension ListComponentViewController: ListElementCollectionViewCellDelegate {
                                                            from: model,
                                                            delegate: self)
     }
+    
+    func longTapGestureActivated(for element: ListNode?, in cell: ListElementCollectionViewCell) {
+        if selectedMultipleItems.isEmpty {
```
Comment (gl-lovekesh) [inline ContentApp/PresentationLayer/Components/ListComponent/ListComponentViewController.swift:409]: will it not work when the list has some items?

File: ContentApp/PresentationLayer/List/Screens/FavoritesViewController.swift [hunk 2]
```diff
@@ -296,6 +298,33 @@ extension FavoritesViewController: ListComponentActionDelegate {
             librariesViewController?.stopLoading()
         }
     }
+    
+    func enabledLongTapGestureForMultiSelection(isShowTabbar: Bool) {
```
Comment (gl-lovekesh) [inline ContentApp/PresentationLayer/List/Screens/FavoritesViewController.swift:302]: The same code is on the [ListViewController.swift](https://github.com/Alfresco/alfresco-mobile-workspace-ios/pull/604/files#diff-e06cd5c832828e4fc3f1acb8da6a43ffd2acee087af9e1258a88cacaea436196)

File: ContentApp/PresentationLayer/List/Screens/ResultViewController.swift [hunk 3]
```diff
@@ -566,6 +566,33 @@ extension ResultViewController: ListComponentActionDelegate {
     func performListAction() {
         // Do nothing
     }
+    
+    func enabledLongTapGestureForMultiSelection(isShowTabbar: Bool) {
```
Comment (gl-lovekesh) [inline ContentApp/PresentationLayer/List/Screens/ResultViewController.swift:570]: Duplicate code as above files.

**Review by gl-lovekesh — CHANGES_REQUESTED** [review]

File: ContentApp/PresentationLayer/Components/ListComponent/ListComponentViewController.swift (hunk 1 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/Components/ListComponent/ListComponentViewController.swift:409]: A long tap gesture would work if the list is empty. After that, a tap gesture would work.

File: ContentApp/PresentationLayer/List/Screens/FavoritesViewController.swift (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/List/Screens/FavoritesViewController.swift:302]: FavoritesViewController has two controllers for files and libraries. This controller is using the delegates of the List Component. I have extended it in the same manner as the current app architecture is using.

File: ContentApp/PresentationLayer/List/Screens/ResultViewController.swift (hunk 3 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/List/Screens/ResultViewController.swift:570]: ResultViewController is used to get the search results. This controller is using the delegates of the List Component. I have extended it in the same manner as the current app architecture is using.

File: ContentApp/PresentationLayer/List/Screens/FavoritesViewController.swift (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/List/Screens/FavoritesViewController.swift:302]: Fixed

File: ContentApp/PresentationLayer/List/Screens/ResultViewController.swift (hunk 3 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/List/Screens/ResultViewController.swift:570]: Fixed

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**